src/routes/todo/TodoPage.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react'
+import { Link, useParams, useNavigate } from 'react-router-dom'
+import { toast } from 'sonner'
+import Todo from '../../components/models/Todo'
+
+type TodoItem = {
+  id: string
+  title: string
+  is_done: boolean
+}
+
+type TodoParams = {
+  id?: string
+  title?: string
+}
+
+const TodoPage: React.FC = () => {
+  const navigate = useNavigate()
+  const params = useParams<TodoParams>()
+  const todos = JSON.parse(localStorage.getItem('TODOS') || '[]') as TodoItem[]
+  const tishTodo = todos.find(t => t.id === params.id)
+  const [todo, setTodo] = useState<Todo | undefined>(tishTodo) // Initialize state
+  const [isEditing, setIsEditing] = useState(false)
+
+  // Getting the status from the matched todo item or defaulting to false
+  const status = todo?.is_done || false
+
+  const toggleTodoStatus = (id: string): void => {
+    const updatedTodos = todos.map(todoItem =>
+      todoItem.id === id
+        ? { ...todoItem, is_done: !todoItem.is_done }
+        : todoItem,
+    )
+    const updatedTodo = updatedTodos.find(t => t.id === id)
+
+    localStorage.setItem('TODOS', JSON.stringify(updatedTodos))
+    if (updatedTodo) setTodo(updatedTodo) // Update the single todo state
+    toast('Status changed.')
+  }
+
+  const deleteTodo = (id: string): void => {
+    const updatedTodos = todos.filter(todoItem => todoItem.id !== id)
+    localStorage.setItem('TODOS', JSON.stringify(updatedTodos))
+    const updatedTodo = updatedTodos.find(t => t.id === id)
+    if (updatedTodo) setTodo(updatedTodo) // Update the single todo state
+    navigate('/')
+  }
+
+  const titleRef = React.useRef<HTMLHeadingElement>(null)
+
+  const handleEdit = () => {
+    const updatedTitle = titleRef.current?.textContent || ''
+    editTodo(todo!.id!, updatedTitle)
+    setIsEditing(false)
+  }
+
+  const editTodo = (id: string, updatedTitle: string): void => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, title: updatedTitle } : todo,
+    )
+    localStorage.setItem('TODOS', JSON.stringify(updatedTodos))
+    const updatedTodo = updatedTodos.find(t => t.id === id)
+    setTodo(updatedTodo)
+    toast.success('Updated.')
+  }
+
+  return (
+    <div className='container mt-5'>
+      <div className='card shadow'>
+        <h2
+          ref={titleRef} // Attach a reference to the DOM element
+          className='card-header text-center bg-primary text-white'
+          contentEditable={isEditing} // Make the content of the <h2> editable based on the `isEditing` state
+        >
+          {todo?.title || 'No Title Provided'}
+        </h2>
+        <div className='card-body'>
+          <p className='card-text'>
+            Todo item with ID:{' '}
+            <span className='font-weight-bold'>
+              {params.id || 'Not Provided'}
+            </span>
+          </p>
+          <p className='card-text'>
+            Status:
```

Comment:
بعنوان پیشنهاد و راه حل:
&nbsp; اینجا لازم هست.

```suggestion
            Status:&nbsp;
```
بهش میگن:
Non-breaking space - Wikipedia
https://en.wikipedia.org/wiki/Non-breaking_space